lib/rules/semi.js

```diff
@@ -233,6 +259,9 @@ module.exports = {
          * @returns {boolean} whether the semicolon is unnecessary.
          */
         function canRemoveSemicolon(node) {
+            if (never && maybeClassFieldAsiHazard(node)) {
+                return false;
+            }
             if (isRedundantSemi(sourceCode.getLastToken(node))) {
                 return true; // `;;` or `;}`
             }
```

Comment:
I think we should switch the order of these two `if`s, to report `get`, `set`, and `static` with `;` at the end of class definitions:

```js
/* eslint semi: ["error", "never"] */

class A {
  static; // false negative?
}
```

Reply:
Fixed.